source/features/clone-branch.tsx

```diff
@@ -0,0 +1,87 @@
+import delegate, {DelegateEvent} from 'delegate-it';
+import gitBranch from 'octicon/git-branch.svg';
+import insertTextTextarea from 'insert-text-textarea';
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getRepoURL} from '../libs/utils';
+
+// Taken from https://stackoverflow.com/questions/3651860/which-characters-are-illegal-within-a-branch-name#comment71900602_3651867
+const invalidBranchName = new RegExp(/^[./]|\.\.|@{|[./]$|^@$|[\s\u0000-\u0020*:?^~\u007F]/); // eslint-disable-line no-control-regex
+async function cloneBranch(event: DelegateEvent<MouseEvent, HTMLAnchorElement>): Promise<void> {
+	const currentTarget = event.delegateTarget;
+	const branchElement = currentTarget.closest<HTMLAnchorElement>('[data-branch-name]');
+	const allExistingBranchNames = select.all('[data-branch-name]').map(element => element.dataset.branchName);
+	let newBranchName = prompt('Enter the new branch name')?.trim();
+
+	while (invalidBranchName.test(newBranchName!) || allExistingBranchNames.includes(newBranchName)) {
+		if (invalidBranchName.test(newBranchName!)) {
+			newBranchName = prompt('Unsupported branch name (https://git-scm.com/docs/git-check-ref-format) \nEnter the new branch name', newBranchName)?.trim();
+		} else {
+			newBranchName = prompt('A branch exists with that name already \nEnter the new branch name', newBranchName)?.trim();
```

Comment:
Good idea, but I think perhaps we should just send the name to the API in the loop. `api.v3` won't throw error if you use `ignoreHTTPStatus: true`. I think something like

```js
let newBranchName = prompt('Enter the new branch name')?.trim();
while (true) {
	const getBranchInfo = await api.v3(...);
	if (getBranchInfo.error) {
		newBranchName = prompt(getBranchInfo.error, '\n Enter the new branch name', newBranchName);
	} else {
		break;
	}
} 
```

Reply:
So I should remove the invalid branch check too?

Reply:
I think so, just keep the `trim`

Reply:
There are 2 api requests.should I do men on the first one? Line this it all does it once?

Reply:
I did it. Changed it a little because your way was no working correctly